OpenSim/Simulation/Model/PhysicalFrame.cpp

```diff
@@ -117,14 +117,14 @@ void PhysicalFrame::scale(const SimTK::Vec3& aScaleFactors)
     for (int i = 0; i< get_WrapObjectSet().getSize(); i++)
         upd_WrapObjectSet().get(i).scale(aScaleFactors);
 
-    SimTK::Vec3 oldScaleFactors;
-    //NewGeometry getDisplayer()->getScaleFactors(oldScaleFactors);
-
-    for (int i = 0; i<3; i++) {
-        oldScaleFactors[i] *= aScaleFactors[i];
+    // Scale the Geometry if any
+    for (int i = 0; i < getNumGeometry(); ++i){
+        const SimTK::Vec3& oldScaleFactor = get_geometry(i).get_scale_factors();
+        //NewGeometry getDisplayer()->getScaleFactors(oldScaleFactors);
+        SimTK::Vec3 newScaleFactor = oldScaleFactor.elementwiseMultiply(aScaleFactors);
+        upd_geometry(i).set_scale_factors(newScaleFactor);
+        // TODO: When we revamp scaling and solve the issue with where scale factors 
+        // are maintained, we need to fix this or remove it completely -Ayman 5/15
+        // Update scale factors for displayer
```

Comment:
The TODO is still relevant because we haven't discussed/decided on what to do when scaling objects attached to a Frame (e.g. does the Frame keep the scale factor or pushes it down, there was a discussion that some Frames may be designated as scalable e.g. at fixed x,y,z wrt ground). I'll update the text to refer to Geometry rather than Displayer. Thanks.